Utils/Dataflow/095_datasetInfoAMI/amiDatasets.py

```diff
@@ -135,36 +178,61 @@ def amiPhysValues(data):
                     p_val = field['$']
                 if p_name and p_val:
                     data[p_name] = p_val
-                    p_name, p_val = None, None
-                    continue
-        return change_key_names(data)
-    except Exception:
-        sys.stderr.write("(WARN) No values found in AMI for dataset '%s'\n"
-                         % data['datasetname'])
+                    break
+        return True
+    except SSLError as e:
+        sys.stderr.write("(ERROR) Failed to process dataset '%s': "
+                         "%r\n" % (data['datasetname'], e))
+        sys.stderr.write("(DEBUG) Traceback start.\n")
+        t, val, trace = sys.exc_info()
+        traceback.print_tb(trace)
+        sys.stderr.write("(DEBUG) Traceback end.\n")
+        return False
+    except Exception as e:
+        sys.stderr.write("(ERROR) Failed to process dataset '%s': "
+                         "%r\n" % (data['datasetname'], e))
+        return False
 
 
 def change_key_names(data):
-    """ Changing parameter names according to PHYS_VALUES dictionary.
+    """ Change parameter names according to ES mapping.
+
+    :param data: data to update
+    :type data: dict
 
-    :param data: JSON string
-    :return: JSON string
+    :return: updated data
+    :rtype: dict
     """
     for item in PHYS_VALUES:
-        if item["ami"] in data:
-            data[item["es"]] = data.pop(item["ami"])
+        if item['ami'] in data:
+            data[item['es']] = data.pop(item['ami'])
     return data
 
 
-def remove_tid(dataset):
-    """ Remove TaskID (_tidXX) part from dataset name.
+def container_name(data):
+    """ Retrieve container name from information about dataset.
+
+    The container name is extracted from dataset name by removing
+    the '_tid...' part.
 
-    As AMI GetPhysicsParamsForDataset works with containers,
-    we construct the container name from each dataset name,
-    removing the _tid{...} part
+    :param data: dataset information
+    :type data: dict
 
-    :param dataset: dataset name
-    :return: dataset name without _tid => container name
+    :return: container name if it was determined successfully, False otherwise
+    :rtype: str or bool
     """
+    if 'datasetname' in data:
+        dataset = data['datasetname']
+    else:
+        sys.stderr.write("(WARN) Required field 'datasetname' not found "
+                         "in data: %r\n" % data)
+        return False
+    if not isinstance(dataset, (str, unicode)):
+        sys.stderr.write("(WARN) Invalid type of 'datasetname' field: "
+                         "expected string, got %s.\n"
+                         "(==) Data containing the field: "
```

Comment:
This information ("containing the field") looks a bit excessive: if we have a value of invalid type in some field, and then output some data as the additional information for this error, it is not difficult to guess that displayed data is where we tried to find our field. Maybe reword it somehow, depending on what it should provide? My variants:
* `"(==) Data: %r\n" % data` *(to output all data)*
* `"(==) Invalid value: %r\n" % dataset` *(to output invalid field value)*

Or maybe some other variant. What do you think?

Reply:
Sure, just "Data:" should do fine. Amended and force-pushed.